src/UrqlClient.re

```diff
@@ -2,14 +2,16 @@ open UrqlTypes;
 type t;
 
 /* Helpers for supporting polymorphic fetchOptions. */
-type fetchOptions =
-  | FetchOpts(Fetch.requestInit)
-  | FetchFn(unit => Fetch.requestInit);
+type fetchOptions('a) =
```

Comment:
🆒this is another instance of the GADT approach, correct? I definitely liked that for `useSubscription`

Reply:
It is! :blush: